backend/server/core/services/startNewChallenge.service.ts

```diff
@@ -45,37 +25,34 @@ function randomArrayShuffle<T>(array: T[]): T[] {
   return array;
 }
 
-export function startNewChallenge(
-  tasks: Task[],
-  challenges: Challenge[],
-  challengeDuration: number = 30,
-  numberOfAchievements: number = Math.floor(challengeDuration / 6)
-): Challenge {
-
-  const actualTasks: ActualTask[] = tasks.map(task => createActualTask(task));
-  const allAchievements: ActualAchievement[] = loadAchievements().map(achievement => createActualAchievement(achievement, actualTasks));
-  const mandatoryAchievements: ActualAchievement[] = allAchievements.filter(
-      (achievement) => achievement.id === 4 || achievement.id === 5
-  );
-  const otherAchievements: ActualAchievement[] = randomArrayShuffle(
-      allAchievements.filter((achievement) => achievement.id !== 4 && achievement.id !== 5)
-  ).slice(0, numberOfAchievements - 2);
-
-  const achievements: ActualAchievement[] = mandatoryAchievements.concat(otherAchievements);
-
-  const tasksOrder: Task[] = randomArrayShuffle(tasks).slice(0, challengeDuration);
-
-  const lastChallengeId = challenges.map((challenge) => challenge.id)
-      .sort((a, b) => a - b)[challenges.length - 1];
-
-  const challenge: Challenge = {
-      id: lastChallengeId + 1,
-      state: StateChallenge.IN_PROGRESS,
-      startDate: new Date(),
-      tasksOrder: tasksOrder,
-      tasksStatus: Constants.PENDING_STATUS,
-      achievementsStatus: Constants.PENDING_STATUS,
-      achievements: achievements,
-  };
-  return challenge;
+export async function startNewChallenge(
+    challengeDuration: number = 30,
+    numberOfAchievements: number = Math.floor(challengeDuration / 6)
+): Promise<ChallengeDocument> {
+    const tasks: TaskDocument[] = await Task.find();
+
+    const allAchievements: AchievementDocument[] = await Achievement.find();
+    const mandatoryAchievements: AchievementDocument[] = allAchievements.filter(
+        (achievement) => achievement.description === 'Complete half of the tasks' ||
```

Comment:
`({ description  }) => description === 'Complete half of the tasks' || description === 'Complete all tasks'`